### DISCUSSION TITLE
Why aren't any of my passwords flagged by a HIBP check?

### DISCUSSION THREAD

**Discussion by phirestalker** [keepassxreboot/keepassxc#8495] [Q&A] [answered]
I received multiple notifications from 2 different services over the past 3 months of breaches with my email address. All of them have said a password was found. I just went to the reports and ran the HIBP check against my passwords, and it came up empty. Does this feature still work? If so, why did it not find any compromised passwords?

**Comment by droidmonkey** [comment, marked as answer]
You'll have to ask the HIBP team. The service works by sending the first part of the hash of your password and the return is a list of all hashes of passwords that match that initial part that have been breached. We then check the full hash against that list. 

It is based off of your password, not your email. If your password wasn't actually breached then it won't be in the list. It shouldn't be in the list if you used a random password.

**Reply by phirestalker** [reply]
OK thanks. I didn't expect it to be based on the email. I meant that they keep saying that there was a password found with my email. I suppose it is possibly the same no longer used password that was leaked long ago.
I will ask them for clarification
When these reports started they used to list the site that the password was for, but now they just say "hey we found your email and some password."

**Reply by phoerious** [reply]
It is based on the password, but if the breached passwords were hashed and salted, there is no way you find yours in the dump.
